pkg/operator/configobserver/apiserver/observe_tlssecurityprofile.go

```diff
@@ -24,23 +24,47 @@ type APIServerLister interface {
 	PreRunHasSynced() []cache.InformerSynced
 }
 
-// ObserveTLSSecurityProfile observes APIServer.Spec.TLSSecurityProfile field and sets
-// the ServingInfo.MinTLSVersion, ServingInfo.CipherSuites fields
-func ObserveTLSSecurityProfile(genericListers configobserver.Listers, recorder events.Recorder, existingConfig map[string]interface{}) (map[string]interface{}, []error) {
-	var (
-		minTlSVersionPath = []string{"servingInfo", "minTLSVersion"}
-		cipherSuitesPath  = []string{"servingInfo", "cipherSuites"}
-	)
+type tlsSecurityObserver struct {
+	minTlSVersionPath []string
+	cipherSuitesPath  []string
+}
+
+var (
+	// minTLSVersionPath provides the path to access minimum serving TLS version in
+	// the observed config based on whether apiServerArguments field should be used
+	minTLSVersionPath = map[bool][]string{
+		false: {"servingInfo", "minTLSVersion"},
+		true:  {"apiServerArguments", "tls-min-version"},
+	}
+	// cipherSuitesPath provides the path to serving cipher suites in the observed
+	// config based on whether apiServerArguments field should be used
+	cipherSuitesPath = map[bool][]string{
+		false: {"servingInfo", "cipherSuites"},
+		true:  {"apiServerArguments", "tls-cipher-suites"},
+	}
+)
+
+// NewTLSSSecurityProfileObserver creates and observer for the APIServer.Spec.TLSSecurityProfile field
+// which sets the ServingInfo.MinTLSVersion and ServingInfo.CipherSuites fields values in the observed
+// configuration (or their apiServerArguments mirrors)
+func NewTLSSSecurityProfileObserver(useAPIServerArguments bool) configobserver.ObserveConfigFunc {
+	observer := &tlsSecurityObserver{
+		minTlSVersionPath: minTLSVersionPath[useAPIServerArguments],
+		cipherSuitesPath:  cipherSuitesPath[useAPIServerArguments],
+	}
+	return observer.observeTLSSecurityProfile
+}
 
+func (o *tlsSecurityObserver) observeTLSSecurityProfile(genericListers configobserver.Listers, recorder events.Recorder, existingConfig map[string]interface{}) (map[string]interface{}, []error) {
```

Comment:
what are we going to do on upgrades, imagine that `existingConfig` was created with `servingInfo`, we rolled out a new version and now we are going to put values into `apiServerArguments`, should we clean up?

Reply:
I think we should, I'll add that

Reply:
actually, since we are not setting that, that will be cleaned up, won't it?